index.html

```diff
@@ -1969,39 +2017,56 @@ <h2>The <dfn>NDEFRecordType</dfn> string</h2>
     </p>
   </section> <!-- visibility & focus: the suspended state -->
 
-  <section><h3>Aborting pending push operation</h3>
+  <section><h3>Clearing the pending push tuple</h3>
   <p>
-    To attempt to <dfn>abort a pending push operation</dfn> on an
+    To <dfn>clear the pending push tuple</dfn> on an
     <a>environment settings object</a>, perform the following steps:
     <ol class=algorithm>
       <li>
         If there is no <a>pending push tuple</a> |tuple|, abort these steps.
       </li>
       <li>
-        Let |writer:NFCWriter| be |tuple|'s writer.
+        Let |writer| be the <a>pending push tuple</a>'s writer.
+      </li>
+      <li>
+        Stop |writer|.[[\PushTimer]] if active.
+      </li>
+      <li>
+        Stop |writer|.[[\PushHandler]] if running.
+      </li>
+      <li>
+        Clear the <a>pending push tuple</a>.
+      </li>
+    </ol>
+  </p>
+</section>
+
+  <section><h3>Aborting pending push operation</h3>
+  <p>
+    To attempt to <dfn>abort a pending push operation</dfn> on an
+    <a>environment settings object</a>, perform the following steps:
+    <ol class=algorithm>
+      <li>
+        If there is no <a>pending push tuple</a> |tuple|, abort these steps.
       </li>
       <li>
         Let |p:Promise| be |tuple|'s promise.
       </li>
       <li>
-        Stop the |writer|'s |timer| if it is active.
+        Let |writer| be the <a>pending push tuple</a>'s writer.
```

Comment:
You are already referring to |tuple| above, so makes more sense to do the same here

Let |writer| be |tuple|'s writer OR {{NFCWriter}} instance

Reply:
OK. Maybe |tuple|'s writer, to follow the style used elsewhere